Automattic/jetpack
# Boost: Replace the site URL by the home URL to cater for subdirectory WordPress install

This PR fixes https://github.com/Automattic/jetpack/issues/21465

Currently, the site speed scores feature is fetching scores based on the URL provided by the Admin class to JavaScript. This URL come from the `get_site_url()` WordPress core method. In 'standard' WordPress site install that method call will return the correct site home URL. However in the setup of a WordPress site that is installed in a subdirectory and has a different site URL and home URL, `get_site_url()` will return the incorrect URL for the context of the site speed scores measuring.

So for a more concrete example:

If a user accesses the site at: www.example.com, but WordPress files are installed at: www.example.com/wordpress, or the root folder of your server's website the difference in the URLs will be:

home_url === "www.example.com"
site_url === "www.example.com/wordpress"

See more about the difference between WordPress home URL and site URL at https://wordpress.stackexchange.com/questions/20294/whats-the-difference-between-home-url-and-site-url

#### Changes proposed in this Pull Request:

* It replaces the `get_site_url()` call with the `get_home_url()` call in order to get the site base URL to cater for sites that have a subdirectory install. Standard site install will not be affected as `get_site_url()` and `get_home_url()` will return the same output.

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
N/A

#### Testing instructions:
* On a 'standard' WordPress install go to the Boost settings page and verify that the site scores are being fetched without error.
* On a WordPress install using subdirectory go to the Boost settings page and verify that the site scores are being fetched without error. 

The changes in this PR is straightforward but if you may want to test it the test environment creation for a non standard WordPress install might not be straightforward itself. Here are the steps I followed to test it about on my local machine:

- Clone the [docker-compose-lamp](https://github.com/sprintcube/docker-compose-lamp) project.
- Install WP in: `/docker-compose-lamp/www/wordpress`.
- From the project root run `mv ./www/wordpress/index.php ./www/index.php`.
- Edit the `./www/index.php` to change the last line to: `require __DIR__ . '/wordpress/wp-blog-header.php';`.
- Start the Docker containers `docker-compose up -d`
- Go to http://localhost:8080/ (`username: root; password: tiger`) for phpMyAdmin and create a new database called `wordpress`.
- From the project root run `cp ./www/wordpress/wp-config-sample.php ./www/wordpress/wp-config.php`.
- Edit the `./www/wordpress/wp-config.php` and setup the constants appropriately. The DB constants will be:
```
define( 'DB_NAME', 'wordpress' );
define( 'DB_USER', 'root' );
define( 'DB_PASSWORD', 'tiger' );
define( 'DB_HOST', 'database' );
define( 'DB_CHARSET', 'utf8' );
```
- Create a tunnel to your local site using `ngrok` localtunnel` or `cloudflared`. Note the domain name provided by one of this service. 
- Add the following in `./www/wordpress/wp-config.php`  replacing `DOMAIN_NAME_PLACEHOLDER` by your domain.
```
define('WP_HOME','http://DOMAIN_NAME_PLACEHOLDER');
define('WP_SITEURL','http://DOMAIN_NAME_PLACEHOLDER/wordpress');
```
- Go to `http://DOMAIN_NAME_PLACEHOLDER` and install the site. The home site should be available at `http://DOMAIN_NAME_PLACEHOLDER` and the WordPress admin site at `http://DOMAIN_NAME_PLACEHOLDER/wordpress/wp-admin`.
- Change the Permalink to post name
- Install Jetpack Boost
- Go to Boost settings page and see that the site speed scores are not being generated.
- Replace `get_site_url()` call with the `get_home_url()` in the `class-admin.php` file as per this PR and refresh the Boost settings page and see that the site speed scores are being generated successfully.

Below is the video of the very two last steps:

https://user-images.githubusercontent.com/927932/141292168-9dad3b38-63a5-45e8-a8a6-65826331c2d3.mp4
